src/libraries/System.Net.Http/tests/FunctionalTests/TelemetryTest.cs

```diff
@@ -614,33 +639,51 @@ public void EventSource_ConnectionPoolAtMaxConnections_LogsRequestLeftQueue()
                             };
                         });
 
-                    await Task.Delay(300);
+                    await WaitForEventCountersAsync(events);
                 });
-                Assert.DoesNotContain(events, ev => ev.EventId == 0); // errors from the EventSource itself
+                Assert.DoesNotContain(events, e => e.Event.EventId == 0); // errors from the EventSource itself
 
-                EventWrittenEventArgs[] starts = events.Where(e => e.EventName == "RequestStart").ToArray();
-                Assert.Equal(3, starts.Length);
-                Assert.All(starts, s => ValidateStartEventPayload(s));
-
-                EventWrittenEventArgs[] stops = events.Where(e => e.EventName == "RequestStop").ToArray();
-                Assert.Equal(3, stops.Length);
-                Assert.All(stops, s => Assert.Empty(s.Payload));
-
-                Assert.DoesNotContain(events, e => e.EventName == "RequestFailed");
+                ValidateStartFailedStopEvents(events, count: 3);
 
                 ValidateConnectionEstablishedClosed(events, version);
 
-                EventWrittenEventArgs requestLeftQueue = Assert.Single(events, e => e.EventName == "RequestLeftQueue");
+                (EventWrittenEventArgs requestLeftQueue, Guid requestLeftQueueId) = Assert.Single(events, e => e.Event.EventName == "RequestLeftQueue");
                 Assert.Equal(3, requestLeftQueue.Payload.Count);
                 Assert.True((double)requestLeftQueue.Payload.Count > 0); // timeSpentOnQueue
                 Assert.Equal(version.Major, (byte)requestLeftQueue.Payload[1]);
                 Assert.Equal(version.Minor, (byte)requestLeftQueue.Payload[2]);
 
+                Assert.Equal(requestLeftQueueId, events.Where(e => e.Event.EventName == "RequestStart").Last().ActivityId);
+
                 ValidateRequestResponseStartStopEvents(events, requestContentLength: null, responseContentLength: 0, count: 3);
 
-                VerifyEventCounters(events, requestCount: 3, shouldHaveFailures: false, requestsLeftQueueVersion: version.Major);
+                ValidateEventCounters(events, requestCount: 3, shouldHaveFailures: false, requestsLeftQueueVersion: version.Major);
             }, UseVersion.ToString()).Dispose();
         }
+
+        private static async Task WaitForEventCountersAsync(ConcurrentQueue<(EventWrittenEventArgs Event, Guid ActivityId)> events)
+        {
+            DateTime startTime = DateTime.UtcNow;
```

Comment:
```suggestion
            long startTime = Environment.TickCount64;
```

It might not matter here since the timeout is 30s, but every time I see `DateTime` used like this, I'm a bit wary.